### DISCUSSION TITLE
HyperSpy Integration with OMERO

### DISCUSSION THREAD

**Discussion by MaadhavKothari** [hyperspy/hyperspy#3303] [General]
Im Maadhav I currently work at Glencoe Software (https://www.glencoesoftware.com) we are the commercial partner of OMERO. 

Im new to the community and interested in integrating HyperSpy with OMERO to extend its functionality beyond the biological space. I'm seeking:

Information on any past attempts or discussions about this integration.
Advice on the feasibility and resources for connecting HyperSpy and OMERO via APIs.
Collaboration or guidance from members experienced with OMERO.
Looking forward to any insights or collaboration. 
FYI OMERO is an open source image data platform https://www.openmicroscopy.org/omero/

Thanks to @CSSFrancis for guiding me to post here. Im looking forward to opening up discussion on what can be possible. 

Best regards,

Maadhav

**Comment by CSSFrancis** [comment]
Hi @MaadhavKothari thank you for starting this discussion!  

> Information on any past attempts or discussions about this integration. Advice on the feasibility and resources for connecting HyperSpy and OMERO via APIs. 

I don't know of any previous attempts at doing this.  The community is large so it's possible that someone has tried it.  We just released a new major release which does include a couple of changes which cleans up our api space.  From a very basic look it seems like there is a 1:1 mapping between the packages( Metadata, axes labels etc.), however, it did seem like OMERO is hard coded with the assumptions that images are 1,2, or 3 dimensions.  Just something to look into. 

A couple of questions/statements:

1. If you are looking for an easy place to start integration [rosettasciio](https://github.com/hyperspy/rosettasciio) is probably the closest thing to your bio-formats package but mostly for microscopy file formats. We recently spun this out for just this kind of purpose!
2. Are you thinking of using `hyperspy` to handle any of the plotting? The hyperspy plotting infrastructure passably fast but more importantly for us very flexible. There are ways we could make it faster by using separate threads for plotting etc. but the developer time for a project like that is currently lacking.  One thing that might be on interest is that hyperspy/rosettasciio [supports](https://github.com/hyperspy/hyperspy/pull/3255) pyodide(and Jupyter-lite) which allows for plotting in a browser etc. 
3. From an instrumentation and features point of view what are you looking to add?  The base hyperspy package no longer has any support for specific scientific problems. Rather hyperspy offers: plotting tools, generic methods for big data processing, curve fitting and metadata management. We have a robust system of tools for more scientific problems listed on our website (https://hyperspy.org). Some of these would be easier (exspy, lumispy, atomap, holospy) to integrate than others (pyxem, kikuchipy) based on the size of that datasets. The first usually have MB to single GB sized datasets while the final two have GB to potentially TB sized datasets. 

> Collaboration or guidance from members experienced with OMERO. Looking forward to any insights or collaboration. FYI OMERO is an open source image data platform https://www.openmicroscopy.org/omero/

From a more practical standpoint, hyperspy has no central funding source currently.  All of the maintainers/developers contribute as some combination of volunteering/ add features related to their work.  This makes larger efforts like this a bit more difficult for us to handle as we don't have a dedicated person reviewing code additions etc.  That isn't to say that things won't get done, just that we move a bit slower. @francisco-dlp and @ericpre might have more insight into this.

**Comment by CSSFrancis** [comment]
@jat255 Just pinging you as well as you might have some insight based on the database side of things with  https://github.com/usnistgov/NexusLIMS
